### PR TITLE
feat: do not break short dot chains

### PR DESCRIPTION
**Pull request by QuadnucYard**
If a dot chain (without comments) starts with an ident, ends with a func call (no other calls inside), and the chain length (estimated from idents length) does not exceed 60% of max-width, the idents will not break into multiple lines.

### CODE REVIEW

**Comment by Enter-tainer** [comment]
> the chain length (estimated from idents length) does not exceed 60% of max-width

Should we consider indentions here? I guess it's not covered at this moment but I think it's ok at least for now.

File: crates/typstyle-core/src/pretty/code_chain.rs [hunk 1]
```diff
@@ -21,25 +27,68 @@ impl<'a> PrettyPrinter<'a> {
...
+
+        // If the chain idents are long enough, do not put them in one line.
+        let estimated_len = ident.get().len()
+            + chain
+                .iter()
+                .skip(1)
+                .map(|child| {
+                    child
+                        .cast::<FieldAccess>()
+                        .map(|field_access| field_access.field().get().len() + 1)
```
Comment (Enter-tainer) [inline crates/typstyle-core/src/pretty/code_chain.rs:74]: `field().len() + 1` 🤔, Are we calculating the length of `.` here ?

File: crates/typstyle-core/src/pretty/code_chain.rs [hunk 2]
```diff
@@ -21,25 +27,68 @@ impl<'a> PrettyPrinter<'a> {
...
+    ) -> Option<ArenaDoc<'a>> {
+        chain.reverse();
+        let func_call = chain.last()?.cast::<FuncCall>()?;
+        let ident = chain[0].cast::<Ident>()?;
+
+        // If the chain idents are long enough, do not put them in one line.
+        let estimated_len = ident.get().len()
+            + chain
+                .iter()
+                .skip(1)
```
Comment (Enter-tainer) [inline crates/typstyle-core/src/pretty/code_chain.rs:70]: quick question: which item are we skipping?

Comment (QuadnucYard) [reply crates/typstyle-core/src/pretty/code_chain.rs:70]: the starting ident

**Comment by QuadnucYard** [comment]
> Should we consider indentions here? I guess it's not covered at this moment but I think it's ok at least for now.

No. So the solution is context-unaware. As long as the chain is estimated to be able to lie in a line, it will not break.

**Review by Enter-tainer — APPROVED** [review]
LGTM

**Pull request merged**